### PR TITLE
Update competitions crud

### LINKED ISSUE #61: Update Competitions CRUD
Update the competition CRUD methods and views to work with the new DB structure as needed.

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Update the competition CRUD stuff to work with the new DB structure. 

#### How should this be manually tested?
Pull down this branch
Run `composer dump-autoload` 
Create a competition, edit a competition, delete a competition, view all competitions. 

#### Any background context you want to provide?

I removed the weird bracket stuff i had in there an we can re-address showing users and user activity in a competition in #57 

#### What are the relevant tickets?
Fixes #61 

cc @DFurnes @weerd @deadlybutter @angaither

### CODE REVIEW

File: resources/views/competitions/partials/_form_competitions.blade.php [hunk 1]
```diff
@@ -1,23 +1,18 @@
 @include('layouts.errors')
 
 <div class="form-item -padded">
-    {!! Form::label('campaign_id', 'Campaign ID:', ['class' => 'field-label']) !!}
-    {!! Form::text('campaign_id', (isset($room->campaign_id)) ? $room->campaign_id : NULL, ['class' => 'text-field']) !!}
+    {!! Form::label('contest_id', 'Contest ID:', ['class' => 'field-label']) !!}
+    {!! Form::text('contest_id', (isset($competition->contest_id)) ? $competition->contest_id : NULL, ['class' => 'text-field']) !!}
```
Comment (weerd) [inline resources/views/competitions/partials/_form_competitions.blade.php]: I mentioned this in @deadlybutter PR too, but can you update this to use:

```php
{{ $competition->contest_id or old('contest_id') }}
```
It'll help with repopulating the field if there was a validation error with what the user entered and also feels much cleaner :) It's a little useful syntactic sugar :)

File: app/Models/Competition.php [hunk 2]
```diff
@@ -12,14 +11,14 @@ class Competition extends Model
      *
      * @var array
      */
-    protected $fillable = ['campaign_id', 'campaign_run_id', 'start_date', 'end_date'];
+    protected $fillable = ['contest_id', 'competition_start_date', 'competition_end_date'];
```
Comment (angaither) [inline app/Models/Competition.php]: should `contest_id` be fillable? isn't it auto increment-able?

Comment (sbsmith86) [inline app/Models/Competition.php]: A competition just references a the contest that it is apart of. This is fillable so we can create competitions and set the contest_id for them.

Comment (weerd) [inline app/Models/Competition.php]: Would the system autofill it at all? Given it's creating a competition for a specified contest? Or do we need to set the contest id each time a competition is created, which is fine too, just more work...

File: resources/views/competitions/partials/_form_competitions.blade.php (hunk 1 above)
Comment (sbsmith86) [inline resources/views/competitions/partials/_form_competitions.blade.php]: @weerd this doesn't seem to work on these date fields when setting the default value. Am I doing something wrong here: 

```{!! Form::date('competition_start_date', $competition->competition_start_date or old($competition->competition_start_date), ['class' => 'text-field']) !!}```

File: app/Models/Competition.php (hunk 2 above)
Comment (deadlybutter) [inline app/Models/Competition.php]: in theory its just 1 line of code when creating the competitions ($competition->contest_id = $contest->id)

I think its fine being fillable

Comment (sbsmith86) [inline app/Models/Competition.php]: I believe, yes, gladiator creates the competitions automatically when the waiting rooms get split ( @deadlybutter is that correct?) But even so, doesn't it need to be able to be mass assigned?

Comment (deadlybutter) [inline app/Models/Competition.php]: yeah thats how its supposed to work, i think its broken right now cause of all the DB updates

Comment (sbsmith86) [inline app/Models/Competition.php]: How are we feeling about this.

Comment (deadlybutter) [inline app/Models/Competition.php]: Im fine with it, @weerd @angaither ?

**Comment by angaither** [comment]
:+1:

File: app/Models/Competition.php (hunk 2 above)
Comment (weerd) [inline app/Models/Competition.php]: Not a huge deal. I think it's fine. If we eventually decide against it, it's an easy fix as @deadlybutter pointed out. Plus it's not the actual contest id being assigned, so should be fine.

**Pull request merged**